test/poolFixtures.hpp

```diff
@@ -493,4 +495,148 @@ TEST_P(umfPoolTest, mallocUsableSize) {
     }
 }
 
+TEST_P(umfPoolTest, umfPoolAlignedMalloc) {
+#ifdef _WIN32
+    // umfPoolAlignedMalloc() is not supported on Windows
+    // TODO: implement support for windows
+    GTEST_SKIP();
+#else  /* !_WIN32 */
+    umf_result_t umf_result;
+    size_t size_of_first_alloc;
+    void *ptr_for_pool = nullptr;
+
+    umf_memory_pool_handle_t pool_get = pool.get();
+    size_of_first_alloc = 2 * 1024 * 1024; // 2MB
+
+    if (!umf_test::isAlignedAllocSupported(pool_get)) {
+        GTEST_SKIP();
+    }
+
+    ptr_for_pool = umfPoolAlignedMalloc(pool_get, size_of_first_alloc,
+                                        utils_get_page_size());
+    ASSERT_NE(ptr_for_pool, nullptr);
+
+    umf_result = umfPoolFree(pool_get, ptr_for_pool);
+    ASSERT_EQ(umf_result, UMF_RESULT_SUCCESS);
+#endif /* !_WIN32 */
+}
+
+TEST_P(umfPoolTest, pool_from_ptr_whole_size_success) {
```

Comment:
Could you add a test case for 8 nested pools and a negative test for 9 nested pools?

Reply:
Done in https://github.com/oneapi-src/unified-memory-framework/pull/1144